docs/support/tutorials/remote-dev.md

```diff
@@ -40,19 +40,100 @@ Also you need to choose a config file where it will save that connection to.
 
 ![Adding a SSH Remote connection](../../img/VSCode_add_connection.png 'Adding SSH connection')
 
+See also the [Remember hosts and advanced settings](https://code.visualstudio.com/docs/remote/ssh#_remember-hosts-and-advanced-settings) section of the VS Code documentation for further details.
+
 After that you can connect to the new host and it will ask you what operation system the target machine has (Puhti is Linux) and also your CSC password.
 
 ![Connecting to a host in VSCode](../../img/VSCode_connect_to_host.png 'Connecting to host')
 
 You are now in Puhti and can open a folder and edit files remotely. Notice that you can also transfer files to Puhti/Mahti by dragging and dropping them to VS Code.
 
-### Running the code using the terminal
+If you prefer to use an ssh private key instead of your password, you can add `-i <path_to_your_keyfile>` to the command shown above. This is mandatory for LUMI, but for Puhti and Mahti you need to first add the corresponding public key to the `~/.ssh/known_hosts` file on the remote system.
 
-You can also run the code in Puhti by opening a terminal from the Terminal tab. Note that it runs the code on a login node by default which is __not meant for demanding computations__. You can open an interactive session with the command
+### Configuring CSC software environments in VS Code Remote SSH
 
+Due to usage of the [module](../../computing/modules.md) system and singularity containers for loading software environments, the VS Code Remote SSH extension is sometimes unable to correctly detect installed software, especially in Python environments. This means that e.g. code completion hints and many other convenient features will not work. Some additional configuration is required in order for VS Code to enable these.
+
+This typically requires running an SSH remote command upon connection and varies depending on your exact use case - details on the exact commands that need to be run are listed below for a few examples. However, in all cases you need to
+
+1. Enable remote commands in the Visual Studio Code Remote-SSH extension:
+   Open the settings screen (`Ctrl ,`) and type "enable remote command" in the search field. The `Enable Remote Command` setting should appear as the first search result. Make sure it is enable (shows the tick mark). ![Enabling Remote Command in Remote SSH extension](../../img/VSCode_enable_remote_command.png).
```

Comment:
"Make sure it is enable**d**" <- "d" missing

Reply:
thanks, fixed